### PR TITLE
ci: build and clean up containers on each PR

### PR DESCRIPTION
**Pull request by cardoe**
Make it possible to test our containers by building and pushing the containers for the life of the PR. Once the PR is closed we'll delete and clean up the containers.

### CODE REVIEW

File: .github/workflows/containers.yaml [hunk 1]
```diff
@@ -149,9 +144,59 @@ jobs:
...
+
+    steps:
+      - name: clean up PR container
+        uses: actions/github-script@v7
+        env:
+          CONTAINER_NAME: '${{ matrix.container }}'
+        with:
+          script: |
+            const container_name = `understack%2F${process.env.CONTAINER_NAME}`;
+            const response = github.rest.packages.getAllPackageVersionsForPackageOwnedByOrg({
```
Comment (skrobul) [inline .github/workflows/containers.yaml:177]: The `getAllPackageVersionsForPackageOwnedByOrg` seems to [be an async](https://github.com/octokit/plugin-rest-endpoint-methods.js/blob/ec6e43c0d80cfbb1fa7768b7dc43368abaa63ad2/src/generated/method-types.ts#L7817) function so should be awaited
   
```suggestion
            const response = await github.rest.packages.getAllPackageVersionsForPackageOwnedByOrg({
```

File: .github/workflows/containers.yaml [hunk 2]
```diff
@@ -149,9 +144,59 @@ jobs:
...
+              package_type: "container",
+              package_name: process.env.CONTAINER_NAME,
+              org: context.repo.owner,
+            });
+
+            const target_tag = `pr-${context.payload.pull_request.number}`;
+            console.log(`Looking for tag ${target_tag}` for container ${container_name}`);
```
Comment (skrobul) [inline .github/workflows/containers.yaml:184]:
```suggestion
            console.log(`Looking for tag ${target_tag} for container ${container_name}`);
```

File: .github/workflows/containers.yaml [hunk 3]
```diff
@@ -149,9 +144,59 @@ jobs:
...
+              version.metadata.container.tags.includes(target_tag)
+            );
+
+            if (matchingVersion) {
+              console.log(`Found tag to delete "${target_tag}":`, matchingVersion);
+              await github.rest.packages.deletePackageVersionForOrg({
+                package_type: "container",
+                package_name: container_name,
+                org: context.repo.owner,
+                package_version_id: pkg_id,
```
Comment (skrobul) [inline .github/workflows/containers.yaml:197]: should that `pkg_id` be replaced with `matchingVersion` ?

File: .github/workflows/containers.yaml [hunk 4]
```diff
@@ -149,9 +144,59 @@ jobs:
...
+            const response = github.rest.packages.getAllPackageVersionsForPackageOwnedByOrg({
+              package_type: "container",
+              package_name: process.env.CONTAINER_NAME,
+              org: context.repo.owner,
+            });
+
+            const target_tag = `pr-${context.payload.pull_request.number}`;
+            console.log(`Looking for tag ${target_tag}` for container ${container_name}`);
+
+            const versions = response.data;
```
Comment (skrobul) [inline .github/workflows/containers.yaml:186]:
```suggestion
            const versions = response.data || [];
```

File: .github/workflows/containers.yaml (hunk 3 above)
Comment (cardoe) [reply .github/workflows/containers.yaml]: I bungled this PR. Thank you for catching all this.

**Comment by cardoe** [comment]
https://github.com/rackerlabs/understack/pkgs/container/understack%2Fironic-nautobot-client/270640358?tag=pr-255 published by this PR

**Pull request merged**